trlx/pipeline/offline_pipeline.py

```diff
@@ -1,29 +1,39 @@
+from copy import deepcopy
+from functools import partial
 from typing import Callable, Iterable, Tuple
 
 import torch
 from datasets import load_dataset
 from torch.nn.utils.rnn import pad_sequence
 from torch.utils.data import DataLoader
 from torchtyping import TensorType
+from transformers import DataCollatorWithPadding
 
 from trlx.data.ilql_types import ILQLBatch, ILQLElement
 from trlx.pipeline import BasePipeline, BaseRolloutStore, register_datapipeline
 
 
 @register_datapipeline
 class OfflinePipeline(BasePipeline):
-    def __init__(self, texts=None):
+    def __init__(self, tokenizer, tokenized):
         super().__init__()
-        self.texts = texts
+        self.tokenizer = deepcopy(tokenizer)
+        if self.tokenizer:
+            self.tokenizer.padding_side = "left"
 
-    def __getitem__(self, index: int):
-        return self.texts[index]
+        self.tokenized = tokenized
+
+    def __getitem__(self, ix: int):
+        return self.tokenized[ix]
 
     def __len__(self) -> int:
-        return len(self.texts)
+        return len(self.tokenized)
 
-    def create_loader(self, batch_size: int, shuffle: bool) -> DataLoader:
-        return DataLoader(self, batch_size=batch_size, shuffle=shuffle)
+    def create_loader(self, batch_size: int) -> DataLoader:
+        collate_fn = (
+            DataCollatorWithPadding(self.tokenizer) if self.tokenizer else torch.vstack
+        )
+        return DataLoader(self, batch_size=batch_size, collate_fn=collate_fn)
 
 
 class OfflineRolloutStorage(BaseRolloutStore):
```

Comment:
Perhaps it makes more sense to construct and load our reward labeled datasets in the rollout storage init? I am unsure but I do think the datasets should be separated from the run scripts